byzcoin/statetrie_test.go

```diff
@@ -71,3 +82,125 @@ func TestStateTrie(t *testing.T) {
 	require.NoError(t, sst.Commit())
 	require.True(t, bytes.Equal(sst.GetRoot(), newRoot))
 }
+
+// TestDarcRetrieval is used as a benchmark here. It stores a lot of darcs
+// that are all linked in a tree-structure, and then uses EvalExprDarc to
+// check whether an identity is valid or not.
+// Use with
+//   go test -v -cpuprofile cpu.prof -run TestDarcRetrieval
+//   go tool pprof -pdf cpu.prof
+func TestDarcRetrieval(t *testing.T) {
+	darcDepth := 3
+	darcWidth := 10
+	entries := 10000
+
+	s := newSer(t, 1, 10*time.Second)
+	defer s.local.CloseAll()
+	s.local.Check = onet.CheckNone
+
+	st, err := s.service().getStateTrie(s.genesis.SkipChainID())
+	require.NoError(t, err)
+	require.NotNil(t, st)
+	require.NotEqual(t, -1, st.GetIndex())
+
+	log.Lvl1("Creating random entries")
+	value := random.Bits(1024, true, random.New())
+	start := time.Now()
+	var scs []StateChange
+	for e := 1; e <= entries; e++ {
+		scs = append(scs, StateChange{
+			StateAction: Create,
+			InstanceID:  random.Bits(256, true, random.New()),
+			ContractID:  ContractDarcID,
+			Value:       value,
+		})
+		if e%100 == 0 {
+			st.StoreAll(scs, 5, CurrentVersion)
+			scs = []StateChange{}
+			log.Print(e, time.Now().Sub(start))
+			start = time.Now()
+		}
+	}
+
+	log.Lvl1("Creating darcs")
```

Comment:
As I understood it, it's mostly two unrelated tests, split it into two functions.

Reply:
No, the measurement of the random entries was just for me to understand how much faster it is to create a slice and then `st.StoreAll` it all at once.

I wanted to measure:
- how long it takes to verify the darcs
- whether entries in the trie change this

Reply:
What about something like:
```
func TestDarcRetrievalWithEmptyTrie()
func TestDarcRetrievalWithFilledTrie()
```
both using a common function which is the second part of the second `TestDarcRetrieval`.

Reply:
doesn't fit my test-case.